### PR TITLE
Update of Plurimath xml engines

### LINKED ISSUE #223: Add spaces before UnitsML in MathML conversion(using Oga)
Add spaces before **Unitsml** in **MathML** output following the instructions mentioned in the https://github.com/metanorma/bipm-si-brochure/issues/222

Process **Oga** nodes in post conversion methods.

### LINKED ISSUE #235: Generalize XML adapters: Oga, Ox
So we are not bound by implementation specific code in the core.

Shale.rb extracts out the XML Adapters, we can learn from their approach.

_Originally posted by @ronaldtse in https://github.com/plurimath/plurimath/issues/223#issuecomment-2014676287_

### PR DESCRIPTION
**Pull request by suleman-uzair**
This PR adds **Plurimath** xml engines wrapper for using Ox and Oga.

closes #235 
closes #223

### CODE REVIEW

File: lib/plurimath/xml_engine/oga/dumper.rb [hunk 1]
```diff
@@ -0,0 +1,96 @@
+module Plurimath
+  module XMLEngine
```
Comment (ronaldtse) [inline lib/plurimath/xml_engine/oga/dumper.rb:2]: Ruby convention of `XmlEngine`?

**Review by ronaldtse — CHANGES_REQUESTED** [review]
Excellent work, thanks!

File: lib/plurimath/xml_engine/oga/dumper.rb (hunk 1 above)
Comment (suleman-uzair) [reply lib/plurimath/xml_engine/oga/dumper.rb:2]: > Ruby convention of `XmlEngine`?

Yes, you are correct, However, I followed the **Oga** file structure to maintain consistency.
But, I can change it if you prefer.

Comment (ronaldtse) [reply lib/plurimath/xml_engine/oga/dumper.rb:2]: Please make the change. Thanks!

Comment (ronaldtse) [reply lib/plurimath/xml_engine/oga/dumper.rb:2]: Bringing other's bad practices into our own is never a good reason 😉

**Review by ronaldtse — APPROVED** [review]
Thank you @suleman-uzair !

**Comment by ronaldtse** [comment]
Need to add documentation on how to use different XML engines.

**Pull request merged**